src/codeql.test.ts

```diff
@@ -89,3 +89,32 @@ test("running a query in a pack", async (t: any) => {
     await rmRF(tmpDir);
   }
 });
+
+test("getting the commit SHA for a database", async (t: any) => {
+  // Note: db-metadata-only.zip is not a real database.
+  // It doesn't contain source code, only a codeql-database.yml file.
+  const testDbZip = path.resolve("testdata/db-metadata-only.zip");
+  const tmpDir = fs.mkdtempSync("tmp");
+  const cwd = process.cwd();
+  process.chdir(tmpDir);
+  try {
+    // Test our fake database. This "database" has a commit SHA in its metadata.
+    await exec("codeql", ["database", "unbundle", testDbZip, "--name=fakeDb"]);
```

Comment:
It's preferable to avoid checking in binary files where possible (although not a hard and fast rule), so you could remove this line and just have the database be a directory rather than a zip file.